src/queue.hpp

```diff
@@ -171,6 +172,16 @@ struct cvk_command_queue : public _cl_command_queue,
         return m_command_pool.free_command_buffer(cmdbuf);
     }
 
+    cvk_buffer* printf_buffer() {
```

Comment:
```suggestion
    cvk_buffer* get_or_create_printf_buffer() {
```

To make the behaviour obvious from the name. We could also have a separate getter for the code running on completion where we could assert the buffer exists. With the current code we could end up creating a buffer in the completion code if we mess up the setup somehow.

Also, note to self: no locking is needed here because we are always holding the queue lock when creating the buffer.